CoreWordCloud/src/au/gov/asd/tac/constellation/views/wordcloud/ui/WordCloudController.java

```diff
@@ -0,0 +1,426 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.views.wordcloud.ui;
+
+import au.gov.asd.tac.constellation.graph.Attribute;
+import au.gov.asd.tac.constellation.graph.Graph;
+import au.gov.asd.tac.constellation.graph.GraphAttribute;
+import au.gov.asd.tac.constellation.graph.GraphElementType;
+import au.gov.asd.tac.constellation.graph.GraphWriteMethods;
+import au.gov.asd.tac.constellation.graph.ReadableGraph;
+import au.gov.asd.tac.constellation.graph.manager.GraphManager;
+import au.gov.asd.tac.constellation.plugins.PluginExecution;
+import au.gov.asd.tac.constellation.plugins.PluginInteraction;
+import au.gov.asd.tac.constellation.plugins.parameters.PluginParameters;
+import au.gov.asd.tac.constellation.plugins.templates.SimpleEditPlugin;
+import au.gov.asd.tac.constellation.utilities.font.FontUtilities;
+import au.gov.asd.tac.constellation.views.wordcloud.phraseanalysis.PhrasiphyContentPlugin;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.SortedSet;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javafx.application.Platform;
+
+/**
+ * Acts as the controller for a WordCloudPane, serving requests from the UI with
+ * reference to a WordCloud object. Listens to the currently active graph to
+ * update the WordCloud object, and in turn, the WordCloudPane.
+ *
+ * @author twilight_sparkle
+ * @author Delphinus8821
+ */
+public class WordCloudController {
+
+    private static final Logger LOGGER = Logger.getLogger(WordCloudController.class.getName());
+    private static WordCloudController instance = null;
+    private WordCloudTopComponent parent;
+    private static final String ATTR_STRING_TYPE = "string";
+    private static final String SELECTED_ATTRIBUTE = "selected";
+    static final String EMPTY_STRING = "";
+    static final ArrayList<String> EMPTY_STRING_LIST = new ArrayList<>(Arrays.asList(EMPTY_STRING));
+    private long attrModCount;
+    private WordCloudPane pane = null;
+    private Graph graph = null;
+    private WordCloud cloud = null;
+    private boolean controllerIsInitialising = false;
+    private final int currentFontSize;
+    
+    private List<String> vertTextAttributes = new ArrayList<>(EMPTY_STRING_LIST);
+    private List<String> transTextAttributes = new ArrayList<>(EMPTY_STRING_LIST);
```

Comment:
might be cheaper to just add the empty string itself rather than a list with only that